rdrf/rdrf/admin.py

```diff
@@ -178,6 +180,37 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('Yes')),
+            ('False', _('No')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_profiles = [profile.id for profile in queryset if profile.activation_key_expired()]
+
+        if self.value() == 'True':
+            return queryset.filter(id__in=expired_profiles)
+        elif self.value() == 'False':
+            return queryset.exclude(id__in=expired_profiles)
+        else:
+            return queryset
+
+
+class CustomRegistrationProfileAdmin(RegistrationAdmin):
+    actions = ['activate_users']
+    list_display = ('user', 'activation_key_expired', 'is_activated')
+    list_filter = [ActivationKeyExpirationListFilter, 'activated']
+
+    @admin.display(description="Activated")
+    def is_activated(self, obj):
+        return obj.activated
```

Comment:
Can be simplified

```suggestion
    list_display = ('user', 'activation_key_expired', 'activated')
    list_filter = [ActivationKeyExpirationListFilter, 'activated']
```

Reply:
Actually, it makes the table inconsistent as it will display True/False and a Tick/Cross, so probably fine as-is